### PR TITLE
Update survival tp53/telomerase analysis

### PR DESCRIPTION
**Pull request by jharenza**
#### What scientific question is your analysis addressing?
This is a followup to 
https://github.com/AlexsLemonade/OpenPBTA-analysis/pull/1208
and 
https://github.com/AlexsLemonade/OpenPBTA-analysis/pull/1212

#### What was your approach?
- Removed samples with missing survival data to get accurate numbers on plots
- Added cancer group to univariate analysis and removed plots, which were not super informative
- Initially, the multivariate analysis did not consider interactions between covariates, so I added them here.
- Added a within cancer group assessment of tp53 scores and telomerase scores, rather than within HGAT or non-HGAT
- Renamed the notebook to describe the contents more accurately
- Did not add README since #1257 is in queue

#### What GitHub issue does your pull request address?
NA

#### Which areas should receive a particularly close look?
I really only updated current code to try to figure out the best way to look at these data and did not do any optimization. 

I think that we can include the coxph plot from chunk 7 in Figure 4, and include some of the tabular statistics for interactions/cancer group tp53/telomerase results/interactions within the text.

Perhaps optimization and/or crisp figure generation is something @runjin326 or @sjspielman can work on? Note: @runjin326 is working on optimizing #1264, so it is possible this analysis can piggyback onto the functions being generated there.

#### Is there anything that you want to discuss further?
Does this workflow make sense?

#### Is the analysis in a mature enough form that the resulting figure(s) and/or table(s) are ready for review?
yes

#### What types of results are included (e.g., table, figure)?
tables, KM curve, hazard plots

#### What is your summary of the results?
**Univariate analysis**
- tp53 scores and telomerase scores are a significant hazard in that higher scores of either are independently associated with poor prognosis (OS)
- HGAT group is also significantly associated with poor prognosis
- EPN, LGG, MB, CRANIO, neurofibroma plexiform, and schwannoma have significant survival benefit (as evidenced by negative estimates and as expected) and DMG has a significant survival hazard.

**Multivariate analysis**
- Higher tp53 scores and higher telomerase scores are each associated with a significant survival hazard (positive estimates) and there is a significant interaction between the two with lower values (negative estimate) having a survival benefit. I am not 100% sure what a negative estimate here means - does it mean both scores are negative or one of the two are negative (which one).
- Here, there is no significant difference between non-HGAT and HGAT (not 100% sure why, but looks like there is too much variance in the HGAT group and it seems like with multiple covariates, this is just not significant).

Within cancer groups:
- Only MB and HGG had significant results.
- For MB - low tp53 and low telomerase scores had a significant survival benefit, and there is a significant interaction between the two. However, I am also not sure how to interpret the positive estimate of this interaction.
- For HGG - higher tp53 and higher telomerase scores were a significant hazard. There is also an interaction, but this has a negative estimate.

#### Reproducibility Checklist

- [x] The dependencies required to run the code in this pull request have been added to the project Dockerfile.
- [x] This analysis has been added to continuous integration.

#### Documentation Checklist

- [ ] This analysis module has a `README` and it is up to date.
- [ ] This analysis is recorded in the table in `analyses/README.md` and the entry is up to date.
- [x] The analytical code is documented and contains comments.

### CODE REVIEW

**Comment by jaclyn-taroni** [comment]
Just adding a link to this comment where interaction was discussed because I keep looking for it and the title for the PR it's on doesn't have a title that makes me think it would be there https://github.com/AlexsLemonade/OpenPBTA-analysis/pull/1216#issuecomment-1011153246

**Comment by sjspielman** [comment]
@jharenza Is there a specific reason the template functions shown in `survival-analysis_template.Rmd` aren't used? The main difference I can see is that this code used survival years instead of days, but this is easy to get around - is there a specific reason we are using years here instead of days? The best way to approach this updated survival notebook and that coming up in #1264 is to use the existing functions and template as much as possible to ensure overall modularity throughout the analysis. 

I'm going to file a PR to your branch that leverages the `util/survival_models.R` functions and we can go from there!

**Comment by jharenza** [comment]
Thanks @sjspielman. I think years were used because the x-axis was too unreadable using days, but that can probably also be fixed using fixed spacing in the plot. I'm not entirely sure about the functions in general- could be that those functions were univariate only and I have a vague memory that there might have been some plot rendering issues.

**Comment by sjspielman** [comment]
>I still need help with is interpretation, though.

Interpreting these interaction models is not going to be particularly straight-forward, beyond just plainly stating which interactions are significant and what coefficients are. This is not something is easily described beyond pure mathematical terms; I don't have "gut sense" beyond that, as it were.

**Comment by jharenza** [comment]
Thanks for the updates @sjspielman - I will take a look in more detail tomorrow, but at first glance, I noticed that in the "TP53 score * Telomerase score, for each cancer group." section, the analyses are not subsetting each group (at least I do not think they are doing it properly). The N being plotted is the same for every group (all 548 samples) and all tables and plots are identical. I had this issue before, but will have to go back and check tomorrow how I had to handle it. Survival analysis is pesky!

File: analyses/survival-analysis/survival-analysis_tp53_telomerase.Rmd
Comment (jaclyn-taroni) [inline analyses/survival-analysis/survival-analysis_tp53_telomerase.Rmd:376]: Pretty sure this is the source of the bug @jharenza mentioned

```suggestion
  fit_plot_save_cox_multivariate(fit_data, 
```

Comment (jharenza) [reply analyses/survival-analysis/survival-analysis_tp53_telomerase.Rmd]: updated this, but also was a bug above, too

**Comment by sjspielman** [comment]
@jaclyn-taroni @jharenza This is now ready for a look. I'm pretty satisfied with this notebook at this stage! Some of the changes I included (in particular a function) can be used in the immune survival notebook I'll get to next, as well.

Summary of the changes I've made:

[(quick jump to rendered notebook)](https://htmlpreview.github.io/?https://raw.githubusercontent.com/AlexsLemonade/OpenPBTA-analysis/36b22380c28e438070c132dd5e3a0df40c964e11/analyses/survival-analysis/survival-analysis_tp53_telomerase.nb.html)

+ There is now a single function added to `util/survival_models.R` that will handle model fitting and exporting as RDS objects, for _both_ univariate and multivariate. As stated, this function can be used for the immune survival notebook coming up next.
+ No longer read in polya telomerase scores since they are removed during processing anyways
+ Added TOC for improved file navigation
+ For the long rank cancer groups model, I added pairwise comparisons among cancer groups, corrected with Bonferroni,+ + I removed plots in favor of printing result tables and exporting fill model fits as RDS. Plots were very busy and not always legible, so I suggest we make plots separately for the manuscript without necessarily making plots in this module.

Note also that the data prep is actually pretty different between the immune scores survival analysis and this one, so we do not need to have a shared data prep script.

File: analyses/survival-analysis/survival-analysis_tp53_telomerase.Rmd
Comment (jharenza) [inline analyses/survival-analysis/survival-analysis_tp53_telomerase.Rmd:140]:
```suggestion
# We want to only include groups that have at least THREE in the DECEASED category. 
```

File: analyses/survival-analysis/survival-analysis_tp53_telomerase.Rmd
Comment (jharenza) [inline analyses/survival-analysis/survival-analysis_tp53_telomerase.Rmd:170]: One thing we also might want to do just as a gut check is run a univariate analysis on molecular subtype for `hgg_group` and `cancer_group` here which have subtypes. This can be in this notebook or can be a part of the immune deconv updates..

File: analyses/survival-analysis/survival-analysis_tp53_telomerase.Rmd
Comment (jharenza) [inline analyses/survival-analysis/survival-analysis_tp53_telomerase.Rmd:181]: all of these RDS files appear NULL when reading them in...

File: analyses/survival-analysis/survival-analysis_tp53_telomerase.Rmd
Comment (jharenza) [inline analyses/survival-analysis/survival-analysis_tp53_telomerase.Rmd:301]:
```suggestion
> ONLY the EPN model shows significant effects for both `tp53_score` and `telomerase_score`, although CIs are fairly wide. The DMG model is marginally significant for `tp53_score` only.
```

Comment (jharenza) [reply analyses/survival-analysis/survival-analysis_tp53_telomerase.Rmd:301]: This is good (re: DMG), and expected!

File: analyses/survival-analysis/survival-analysis_tp53_telomerase.Rmd
Comment (sjspielman) [reply analyses/survival-analysis/survival-analysis_tp53_telomerase.Rmd]: good catch!

File: analyses/survival-analysis/survival-analysis_tp53_telomerase.Rmd
Comment (sjspielman) [reply analyses/survival-analysis/survival-analysis_tp53_telomerase.Rmd:170]: I think since the immune deconv is already looking at molecular signatures an extent, we can do it there.

File: analyses/survival-analysis/survival-analysis_tp53_telomerase.Rmd
Comment (sjspielman) [reply analyses/survival-analysis/survival-analysis_tp53_telomerase.Rmd:181]: Oh no, I will have a look..

Comment (sjspielman) [reply analyses/survival-analysis/survival-analysis_tp53_telomerase.Rmd:181]: This was a silly typo! Fix incoming!

File: analyses/survival-analysis/survival-analysis_tp53_telomerase.Rmd
Comment (jharenza) [reply analyses/survival-analysis/survival-analysis_tp53_telomerase.Rmd:170]: sounds good!

**Comment by jharenza** [comment]
this looks good to me now - ready for re-review @jaclyn-taroni !

File: analyses/survival-analysis/survival-analysis_tp53_telomerase.Rmd
Comment (jaclyn-taroni) [inline analyses/survival-analysis/survival-analysis_tp53_telomerase.Rmd:298]:
```suggestion
### Additive models: `tp53_score+telomerase_score` _for each cancer group_
```

File: analyses/survival-analysis/survival-analysis_tp53_telomerase.Rmd
Comment (jaclyn-taroni) [inline analyses/survival-analysis/survival-analysis_tp53_telomerase.Rmd:146]: Is this comment correct? I am confused.

File: analyses/survival-analysis/survival-analysis_tp53_telomerase.Rmd
Comment (jaclyn-taroni) [inline analyses/survival-analysis/survival-analysis_tp53_telomerase.Rmd:271]: If it's easy enough to extract just the p-value from what `fit_save_model()`, you could put this after the code chunk and use  p = `r <variable with the p-value in it>`. Just a bit of future-proofing for thought.

**Review by jaclyn-taroni — APPROVED** [review]
I had a few questions and suggestions, but nothing significant. Based on our offline (online-but-not-GitHub) conversations, I expect you did functional testing to ensure that the move to the new function didn't impact the results. I'm approving with this assumption!

File: analyses/survival-analysis/survival-analysis_tp53_telomerase.Rmd
Comment (sjspielman) [reply analyses/survival-analysis/survival-analysis_tp53_telomerase.Rmd:238]: The `survival_analysis()` function was originally created to do the recoding within the function and expects "LIVING" and "DECEASED" - 
https://github.com/AlexsLemonade/OpenPBTA-analysis/blob/51e4fca33eaa306ec7e63291217dc03551bb1e13/analyses/survival-analysis/util/survival_models.R#L97 

I suppose the code would still work if this were already recoded, but I thought it would be preferable to follow with the precedent that was set in the [template Rmd](https://github.com/AlexsLemonade/OpenPBTA-analysis/blob/51e4fca33eaa306ec7e63291217dc03551bb1e13/analyses/survival-analysis/survival-analysis_template.Rmd) not to recode.

File: analyses/survival-analysis/survival-analysis_tp53_telomerase.Rmd
Comment (sjspielman) [reply analyses/survival-analysis/survival-analysis_tp53_telomerase.Rmd:146]: The comment is explaining line 147 - `fill = 0`

Comment (sjspielman) [reply analyses/survival-analysis/survival-analysis_tp53_telomerase.Rmd:146]: This part of the code is still counting the number of events, and `NA`s will occur when spreading if there is no recorded event. `indicating 0 deceased or 0 living` is referring to number of participants, not the recoding.

I'll make sure this is doing what I think it is and rephrase a little.

Comment (jaclyn-taroni) [reply analyses/survival-analysis/survival-analysis_tp53_telomerase.Rmd:146]: My issue is with the phrasing. Using "no events" seems less confusing to me.

File: analyses/survival-analysis/survival-analysis_tp53_telomerase.Rmd
Comment (sjspielman) [reply analyses/survival-analysis/survival-analysis_tp53_telomerase.Rmd]: Done.

**Comment by sjspielman** [comment]
This has passed CI successfully before merging master in, so this PR be merged in before checks are complete.

**Pull request merged**
